### ISSUE TITLE
HeroG Testing feedback testnet Bitcoin token transfers

### ISSUE THREAD

**Issue by Hero-Gamer** [blockstack/blockstack-app#3153]
**Minor:** 
- 1. Don't understand the purpose of the receive button when I can just click copy address from the main screen, it's an extra unnecessary step as of now. I want as few clicks as possible.
- 2. Leads me to suggest, on the main wallet screen there should be a logo for Stacks, and a logo for Bitcoin next to each address, so people can simply just copy the right address accordingly.
- 3. When sending to "another account" of yourself's own wallet, one letter is off at the end from displaying the full complete bitcoin address, why not just shrink the text size or increase the box, so the full bitcoin address can be fully visible. 

**Major:**
- 4. on the SEND screen, inputting the number of BTC at the top doesn't seem intuitive at the moment, I was clicking on the bitcoin logo box to attempt to input the BTC amount there. Feels like maybe that's a more natural intuitive place for people to input the BTC amount
- 5. When choosing the amount, I cannot see HOW MUCH I BTC in total have in my wallet, nor can I see btc left if I enter to send certain amount, NOR can I see the fee on the same screen. I think every detail can be displayed on the send screen so I can understand how much I'm sending in context of how much I actually have. 
- 6. When click SEND ALL, it doesn't tell me why I cannot send all 0.0006 bitcoin off, I understand why, it's due to fee, but no fee info is displayed, so it is confusing if no breakdown details when operating for a user. 
- 7. At the last page, the preview page, it DOES NOT tell me the TOTAL BTC I'm actually gonna be sending, it tells me what I'm sending, and the fee separately, you MUST have a total amount i.e. "amount wanna send" + "fee" = Total BTC transaction amount, this amount should match the "pending" tx amount send after you click send. 
 
**Trivial:**
- 1. Why use https://blockstream.info/testnet explorer, is there no better one?

**Comment by kyranjamie** [comment]
Thanks for the feedback @Hero-Gamer, some of these are already being addresses. @mica000 @markmhx to review the rest.

**Comment by markmhendrickson** [comment]
Much thanks for all this @Hero-Gamer !

> Don't understand the purpose of the receive button when I can just click copy address from the main screen, it's an extra unnecessary step as of now. I want as few clicks as possible.

Soon we're going to show a QR for receipt when you click the "Receive" button as well as a special Taproot address for depositing new inscriptions. As it stands, you're right it doesn't provide a lot of extra value but it's a component that'll soon get extended for more.

> Leads me to suggest, on the main wallet screen there should be a logo for Stacks, and a logo for Bitcoin next to each address, so people can simply just copy the right address accordingly.

Interesting idea. @mica000 can consider

> When sending to "another account" of yourself's own wallet, one letter is off at the end from displaying the full complete bitcoin address, why not just shrink the text size or increase the box, so the full bitcoin address can be fully visible.

Mind sharing a screenshot of this?

> on the SEND screen, inputting the number of BTC at the top doesn't seem intuitive at the moment, I was clicking on the bitcoin logo box to attempt to input the BTC amount there. Feels like maybe that's a more natural intuitive place for people to input the BTC amount

Good feedback. Perhaps we can simply make this logo box trigger focusing the input box in the very short term, and @mica000 can think through further enhancements so it's clearer:

<img width="499" alt="image" src="https://user-images.githubusercontent.com/28991/219304859-3cae650f-29f5-4841-87ee-d9cf2362d738.png">


> When choosing the amount, I cannot see HOW MUCH I BTC in total have in my wallet, nor can I see btc left if I enter to send certain amount, NOR can I see the fee on the same screen. I think every detail can be displayed on the send screen so I can understand how much I'm sending in context of how much I actually have.

Agreed. We're tracking "how much" with https://github.com/hirosystems/stacks-wallet-web/issues/3143 and I'll write a note there about it'd be helpful to show "how much remaining" as well. Fees will be handled in the next screen of the flow given the variability of trying to handle them on the same screen. Separating is common practice among Bitcoin wallets.

> When click SEND ALL, it doesn't tell me why I cannot send all 0.0006 bitcoin off, I understand why, it's due to fee, but no fee info is displayed, so it is confusing if no breakdown details when operating for a user.

Yep we're working on improving fee handling in general. We should have another build for further testing later today.

> At the last page, the preview page, it DOES NOT tell me the TOTAL BTC I'm actually gonna be sending, it tells me what I'm sending, and the fee separately, you MUST have a total amount i.e. "amount wanna send" + "fee" = Total BTC transaction amount, this amount should match the "pending" tx amount send after you click send.

Same here, we're working on this at the moment.

> Why use https://blockstream.info/testnet explorer, is there no better one?

Any other one you'd recommend specifically? It was just a quick choice on our side; there's no strong preference for it.

**Comment by mica000** [comment]
Thank you so much for the detailed feedback @Hero-Gamer!

> Don't understand the purpose of the receive button when I can just click copy address from the main screen, it's an extra unnecessary step as of now. I want as few clicks as possible.
> 
> Leads me to suggest, on the main wallet screen there should be a logo for Stacks, and a logo for Bitcoin next to each address, so people can simply just copy the right address accordingly.


Could you test this prototype and see if this could improve the problem of copying the addresses in a quicker manner?

<img width="1529" alt="Screenshot 2023-02-16 at 12 25 13" src="https://user-images.githubusercontent.com/119523541/219352556-463f322c-91f2-46c8-bcfa-b4299e8e3edf.png">

https://www.figma.com/proto/ooIjZrRmCDeMj1q6pcOEnR/Extension-wallet?page-id=0%3A1&node-id=42%3A4336&viewport=3405%2C3905%2C0.61&scaling=min-zoom&starting-point-node-id=42%3A4336&show-proto-sidebar=1

**Comment by mica000** [comment]
> on the SEND screen, inputting the number of BTC at the top doesn't seem intuitive at the moment, I was clicking on the bitcoin logo box to attempt to input the BTC amount there. Feels like maybe that's a more natural intuitive place for people to input the BTC amount

That's interesting @Hero-Gamer! Thanks for flagging

@markmhx! @fbwoolf Here is a recording with comments on what needs to be improved so the UX on typing is clearer:

https://user-images.githubusercontent.com/119523541/219353898-e161c2ce-2584-437d-8312-32d239f11e8d.mov


Designs: https://www.figma.com/file/OSQocEEJnfcMkniEl8u3U8/%231483---Initial-Bitcoin-support?node-id=1101%3A56132&t=Shl5eSB4xRFAbQkH-4

**Comment by mica000** [comment]
> When choosing the amount, I cannot see HOW MUCH I BTC in total have in my wallet, nor can I see btc left if I enter to send certain amount, NOR can I see the fee on the same screen. I think every detail can be displayed on the send screen so I can understand how much I'm sending in context of how much I actually have.

Agree! we are fixing this with this addition on the "Send all" CTA
<img width="511" alt="Screenshot 2023-02-16 at 12 48 31" src="https://user-images.githubusercontent.com/119523541/219357152-1cee14fd-5d9f-4fa7-a483-eaa0c0ede25a.png">

**Comment by mica000** [comment]
> When click SEND ALL, it doesn't tell me why I cannot send all 0.0006 bitcoin off, I understand why, it's due to fee, but no fee info is displayed, so it is confusing if no breakdown details when operating for a user.
> 
> At the last page, the preview page, it DOES NOT tell me the TOTAL BTC I'm actually gonna be sending, it tells me what I'm sending, and the fee separately, you MUST have a total amount i.e. "amount wanna send" + "fee" = Total BTC transaction amount, this amount should match the "pending" tx amount send after you click send.

What do you think about these fixes @Hero-Gamer @markmhx ?



https://user-images.githubusercontent.com/119523541/219362167-e6184add-1514-4cc0-ba95-1403e3a15479.mov

**Comment by Hero-Gamer** [comment]
After using the new wallet v4 to send STX, I feel the most logical flow from top to bottom is. 
1. enter / paste an STX address
2. enter STX amount

Currently the flow is, I paste in the address, then my eyes & hand have to somehow scan back up to the top of the window to enter the amount. This is why it is not quite flowing smoothly from my perspective. 

I haven't looked into the BTC transfer flow yet mentioned above. Will come back later.

**Comment by markmhendrickson** [comment]
@fabric-8 @mica000 assigning you for context as you work on transaction signing UX improvements in general

We can tackle other feedback here (e.g. regarding balances, receive, etc) when we look at those flows
